chain/client/src/chunk_validation.rs

```diff
@@ -64,50 +70,38 @@ impl ChunkValidator {
 
         let network_sender = self.network_sender.clone();
         let signer = self.my_signer.clone().unwrap();
-        let runtime_adapter = self.runtime_adapter.clone();
-        rayon::spawn(move || match validate_chunk(&state_witness, runtime_adapter.as_ref()) {
-            Ok(()) => {
-                tracing::debug!(
-                    target: "chunk_validation",
-                    chunk_hash=?chunk_header.chunk_hash(),
-                    block_producer=%block_producer,
-                    "Chunk validated successfully, sending endorsement",
-                );
-                let endorsement_to_sign = ChunkEndorsementInner::new(chunk_header.chunk_hash());
-                network_sender.send(PeerManagerMessageRequest::NetworkRequests(
-                    NetworkRequests::ChunkEndorsement(ChunkEndorsementMessage {
-                        endorsement: ChunkEndorsement {
-                            account_id: signer.validator_id().clone(),
-                            signature: signer.sign_chunk_endorsement(&endorsement_to_sign),
-                            inner: endorsement_to_sign,
-                        },
-                        target: block_producer,
-                    }),
-                ));
-            }
-            Err(err) => {
-                tracing::error!("Failed to validate chunk: {:?}", err);
+        rayon::spawn(move || {
+            let parent_span =
```

Comment:
FWIW this span will be orphaned as it is the first one in a new thread. What you may want is to create this span outside of the `rayon::spawn` so that it picks up its own parent span from the environment, and for all the spans inside specify the parent explicitly via `parent: parent_span` argument.